internal/lib/lib.go

```diff
@@ -1710,6 +1710,35 @@ func GetDefaultSecretForRoutes() string {
 	return DefaultRouteCert
 }
 
+func ValidateSvcforClass(key string, svc *corev1.Service) bool {
+	if svc != nil {
+		if svc.Spec.LoadBalancerClass == nil {
+			if isAviDefaultLBController() {
+				utils.AviLog.Infof("key: %s, msg: LoadBalancerClass is not specified for LB service %s but ako.vmware.com/avi-lb is default loadbalancer controller", key, svc.ObjectMeta.Name)
+				return true
+			} else {
+				utils.AviLog.Warnf("key: %s, msg: LoadBalancerClass is not specified for LB service %s and ako.vmware.com/avi-lb is not default loadbalancer controller", key, svc.ObjectMeta.Name)
+				return false
+			}
+		} else {
+			if *svc.Spec.LoadBalancerClass != AviIngressController {
+				utils.AviLog.Warnf("key: %s, msg: LoadBalancerClass for LB service %s is not ako.vmware.com/avi-lb", key, svc.ObjectMeta.Name)
+				return false
+			} else {
+				utils.AviLog.Infof("key: %s, msg: LoadBalancerClass for LB service %s is ako.vmware.com/avi-lb", key, svc.ObjectMeta.Name)
+				return true
+			}
+		}
+	}
+	utils.AviLog.Warnf("key: %s, msg: Could not find service for LBClass Validation")
+	return false
+}
+
+func isAviDefaultLBController() bool {
```

Comment:
This will fetch OS env each time. So lets set it once during AKO bootup and get it from Global variable. Use akoControlConfig struct for this.

Reply:
I've added relevant variable in AKOControlConfig and util functions to populate and refer to the same